src/json-patch-duplex.ts

```diff
@@ -290,11 +290,15 @@ module jsonpatch {
   }
 
   function deepClone(obj:any) {
-    if (typeof obj === "object") {
-      return JSON.parse(JSON.stringify(obj)); //Faster than ES5 clone - http://jsperf.com/deep-cloning-of-objects/5
-    }
-    else {
-      return obj; //no need to clone primitives
+    switch (typeof obj) {
+      case "object":
+        return JSON.parse(JSON.stringify(obj)); //Faster than ES5 clone - http://jsperf.com/deep-cloning-of-objects/5
+
+      case "undefined":
+        return null; //this is how JSON.stringify behaves for array items
```

Comment:
In fact it is covered by `undefined` check at 
https://github.com/Starcounter-Jack/JSON-Patch/blob/0965f2fe84e04e8a3d18eaf3d1d4abb8c6bea1fc/src/json-patch-duplex.ts#L409
and 
https://github.com/Starcounter-Jack/JSON-Patch/blob/0965f2fe84e04e8a3d18eaf3d1d4abb8c6bea1fc/src/json-patch-duplex.ts#L431

So the answer is: "It does nullify `undeifined`s, but we never call `deepClone` it on regular object properties"